src/handwritten/_linux_installation_commands.py

```diff
@@ -40,25 +62,23 @@ def _get_version_rhel(dist_version: str) -> str:
 rpm_daqmx_version_command = ["rpm", "-q", "ni-daqmx"]
 
 
+@dataclass
+class DistroInfo:
+    get_distro_version: Callable[[str], str]
+    get_daqmx_version: Tuple[str, ...]
+    install_commands: Tuple[Tuple[str, ...], ...]
```

Comment:
The variables above use lists, not tuples.
```suggestion
    get_daqmx_version: List[str]
    install_commands: List[List[str]]
```

Reply:
BTW, I expect `poetry run mypy` to tell you this.

Reply:
It may not since mypy is about being type safe as opposed to hard-typed.
(i.e, if all references to this do not attempt to do the things that you can with a list but not a tuple like change them, the mypy should not complain about those)

Reply:
If people run the lint tools, then I don't have to leave as many comments about things that the lint tools would have caught.

If people don't run the lint tools, then I may or may not catch these things, but sometimes I do.

```
PS C:\dev\nidaqmx-python> poetry run mypy generated/nidaqmx
generated\nidaqmx\_linux_installation_commands.py:74: error: Argument 2 to "DistroInfo" has incompatible type "list[str]"; expected "tuple[str, ...]"  [arg-type]
generated\nidaqmx\_linux_installation_commands.py:74: error: Argument 3 to "DistroInfo" has incompatible type "list[list[str]]"; expected "tuple[tuple[str, ...], ...]"  [arg-type]
generated\nidaqmx\_linux_installation_commands.py:76: error: Argument 2 to "DistroInfo" has incompatible type "list[str]"; expected "tuple[str, ...]"  [arg-type]
generated\nidaqmx\_linux_installation_commands.py:76: error: Argument 3 to "DistroInfo" has incompatible type "list[list[str]]"; expected "tuple[tuple[str, ...], ...]"  [arg-type]
generated\nidaqmx\_linux_installation_commands.py:78: error: Argument 2 to "DistroInfo" has incompatible type "list[str]"; expected "tuple[str, ...]"  [arg-type]
generated\nidaqmx\_linux_installation_commands.py:78: error: Argument 3 to "DistroInfo" has incompatible type "list[list[str]]"; expected "tuple[tuple[str, ...], ...]"  [arg-type]
Found 6 errors in 1 file (checked 73 source files)
```

Reply:
BTW if you want to accept both list and tuple then using Iterable[str], but note that this also accepts a plain str, since a str is an Iterable[str] of characters.

Reply:
> If people don't run the lint tools, then I may or may not catch these things, but sometimes I do.

Is this not part of the PR checks?

> BTW if you want to accept both list and tuple then use Iterable[str], but note that this also accepts a plain str, since a str is an Iterable[str] of characters.

It also makes mypy complain about asking for the length because (while strings and tuples are iterable) not all iterables are a fixed/known length.

Going down this tangent further, we can put `Collection[str]` or `Union[List[str], Tuple[str]]`, both of these are "sized iterables of str"; however, the latter is more constraining.

(some context on why this is coming up, when I buddy coded this with @dharaniprakashkm, I commented that these should be Tuples because they should not be changing - and so the code I typed and sent via chat hinted for Tuples)

Reply:
> Is this not part of the PR checks?

Yes, but these didn't run because there is a merge conflict.

> It also makes mypy complain about asking for the length because (while strings and tuples are iterable) not all iterables are a fixed/known length.

Good point.

> (some context on why this is coming up, when I buddy coded this with @dharaniprakashkm, I commented that these should be Tuples because they should not be changing - and so the code I typed and sent via chat hinted for Tuples)

I see. Either list or tuple is fine with me, as long as the data matches the type hints.